sidekick/lib/src/commands/init_command.dart

```diff
@@ -134,123 +204,127 @@ class InitCommand extends Command {
-          'https://github.com/passsy/flutter_wrapper\n\n'
-          'This allows you to use the `$cliName dart` and `$cliName flutter` commands\n');
-      final confirmFlutterwInstall = dcli.confirm(
-        'Install flutterw?',
-        defaultValue: false,
-      );
-      if (confirmFlutterwInstall) {
-        await installFlutterWrapper(entrypointDir);
-      }
-    }
-
     // Download the bundled dart runtime for the CLI
     final bundledDart = (SidekickDartRuntime(cliPackage)..download()).dart;
 
-    // TODO add --offline flag that does not upgrade anything
     // make sure sidekick_core is up-to-date
-    bundledDart(
-      ['pub', 'upgrade', 'sidekick_core'],
-      workingDirectory: cliPackage,
-    );
+    final errorCapture = Progress.capture();
+    try {
+      bundledDart(
+        ['pub', 'upgrade', 'sidekick_core'],
+        workingDirectory: cliPackage,
+        progress: errorCapture,
+      );
+    } catch (e) {
+      // print only in case of error
+      printerr(red(errorCapture.lines.join('\n')));
+      rethrow;
+    }
 
     bundledDart(
       ['format', cliPackage.path],
       progress: dcli.Progress.printStdErr(),
     );
-    print(
-      green('Successfully generated ${cliName}_sidekick 🎉'),
-    );
   }
-}
 
-/// Initializes git via `git init` in [directory]
-Future<void> gitInit(Directory directory) async {
-  final bool inGitDir = Process.runSync(
-        'git',
-        ['rev-parse', '--git-dir'],
-        workingDirectory: directory.path,
-      ).exitCode ==
-      0;
-  if (inGitDir) {
-    // no need to initialize
-    return;
+  /// Asks user to install flutterw_sidekick_plugin when a Flutter package is detected
+  void installFlutterWrapper(_InitInputs inputs) {
+    final entrypoint = inputs.projectRoot.file(inputs.cliName);
+
+    final flutterPackages = [
+      if (inputs.mainProject != null) inputs.mainProject!,
+      ...inputs.packages
+    ].filter((package) => package.isFlutterPackage).toSet();
+
+    if (flutterPackages.isNotEmpty) {
+      print(
+        'Sidekick detected ${flutterPackages.length} Flutter package(s) in your project.\n',
+      );
+      print(
+        "It's recommended to bind an exact Flutter version to your project "
+        "and share the same version with your coworkers and CI. There are two ways to accomplish this:\n"
+        " - Use FVM (https://fvm.app/)\n"
+        " - Use flutterw (https://github.com/passsy/flutter_wrapper)\n",
+      );
+
+      print(
+        '${dcli.green('Do you want pin the Flutter version of this project with flutterw?')}\n'
+        'This will download the Flutter SDK ~900mb',
+      );
+      final confirmFlutterwInstall = dcli.confirm(
+        'Install flutterw_sidekick_plugin?',
+        defaultValue:
+            env['SIDEKICK_INIT_APPROVE_FLUTTERW_INSTALL'] == 'true' || false,
+      );
+      if (confirmFlutterwInstall) {
+        "${entrypoint.path} sidekick plugins install flutterw_sidekick_plugin"
```

Comment:
Nice 😊
This is printing a lot of noise because we run the entrypoint for the first time, so all the output for compiling it and getting the dependencies etc. is printed.

Before running `"${entrypoint.path} sidekick plugins install flutterw_sidekick_plugin"`, we could run `dcli.start(entrypoint.path, progress: Progress.devNull());` to get rid of a lot of the noise.
Afterwards run `"${entrypoint.path} sidekick plugins install flutterw_sidekick_plugin".run;` as is to print the output of the `flutterw_sidekick_plugin` installation


This output is fine until `Install flutterw_sidekick_plugin?`
![image](https://user-images.githubusercontent.com/39117631/215466073-f60a66e5-6583-4444-833d-16a955c3c077.png)

Remove this noise
![image](https://user-images.githubusercontent.com/39117631/215466216-b01bd834-6687-47d4-bfb7-df5ffeac681d.png)

Output from `Installing flutterw_sidekick_plugin` is fine
![image](https://user-images.githubusercontent.com/39117631/215466304-5be22bd3-ab2e-42cb-8381-9fa4dc8078f6.png)